src/App/Utilities/AppHelpers.cs

```diff
@@ -473,21 +472,21 @@ public static async Task LogOutAsync(string userId, bool userInitiated = false)
                 folderService.ClearAsync(userId),
                 collectionService.ClearAsync(userId),
                 passwordGenerationService.ClearAsync(userId),
-                deviceActionService.ClearCacheAsync());
-            if (userInitiated)
-            {
-                await Task.WhenAll(
-                    syncService.SetLastSyncAsync(DateTime.MinValue),
-                    tokenService.ClearTokenAsync(userId),
-                    cryptoService.ClearKeysAsync(userId),
-                    settingsService.ClearAsync(userId),
-                    vaultTimeoutService.ClearAsync(userId),
-                    stateService.ClearAsync(userId));
-                if (await stateService.GetActiveUserIdAsync() != null)
-                {
-                    var messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
-                    messagingService.Send("switchedAccount");
-                }
+                deviceActionService.ClearCacheAsync(),
+                tokenService.ClearTokenAsync(userId),
+                cryptoService.ClearKeysAsync(userId),
+                settingsService.ClearAsync(userId),
+                vaultTimeoutService.ClearAsync(userId),
+                stateService.LogoutAccountAsync(userId, userInitiated));
+
+            // check if we switched accounts automatically
+            if (userInitiated && await stateService.GetActiveUserIdAsync() != null)
+            {
+                var messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
+                messagingService.Send("switchedAccount");
+                
+                var platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+                platformUtilsService.ShowToast("info", null, AppResources.AccountSwitchedAutomatically);
```

Comment:
If possible I think it'd be better to tell the user which account is the current one on the message as well, like "Switched to next available account: myaccount@domain.com" or directly "Switched to myaccount@domain.com".
And is it ok for it to be here instead of inside the logic of the subscription of "switchedAccount" `messagingService`? Because the logic makes async stuff and that is not awaited on `messagingService.Send("switchedAccount");` so the message could be shown before the account is actually switched. Furthermore, if any other place calls "switchedAccount" would need to add the toast thus IMO it makes more sense to put it in the `SwitchedAccountAsync` of the `App.xaml.cs`

Reply:
My only concern with including the email address is the potential for really long addresses being crammed into a toast.  I'll run it by design though.

I see what you're saying on the second point - taking another look now